server/Controllers/GameControllers/snake.js

```diff
@@ -1,12 +1,115 @@
+const { SNAKE } = require('./gameConstants');
+
 class Snake {
-  constructor(playerId, initialPositionX, initialPositionY,
-    initialDirection, snakeColor, initialSnakeLength) {
+  /**
+   * @param  {string} playerId
+   * @param  {Object} initialProperties initial values for snake body and motion.
+   * @param  {Object} game instance of game class for status of current game. 
+   */
+  constructor(playerId, initialProperties, game) {
     this.ownerId = playerId;
-    this.head = { x: initialPositionX, y: initialPositionY };
-    this.direction = initialDirection;
-    this.color = snakeColor;
-    this.bodyLength = initialSnakeLength;
+    this.head = { x: initialProperties.position.x, y: initialProperties.position.y };
+    this.direction = initialProperties.direction;
+    this.velocity = initialProperties.velocity;
+    this.color = initialProperties.color;
+    this.length = initialProperties.length;
+    this.eatFood = game.eatFood;
+    this.rivalBody = game.rivalBody;
+    this.gameInterval = game.gameInterval; // interval variable, if snake collides we will clear this interval to stop the game
+    this.life = SNAKE.LIFE;
+    this.lost = false;
+    this.xAxisVelocity = 0; // the x axis movement of snake in each interval
+    this.yAxisVelocity = 0; // the y axis movement of snake in each interval
     this.bodyCoordinates = [];
+    this.createInitialSnakeBody(); // populate initial body of snake according to given length.
+  }
+
+  /**
+   * This function generates snake body coordinates from the given starting head coordinates
+   */
+  createInitialSnakeBody() {
+    switch (this.direction) {
+      case 'down': {
+        this.yAxisVelocity = this.velocity;
+        break;
+      }
+      case 'up': {
+        this.yAxisVelocity = -this.veloctiy;
+        break;
+      }
+      case 'left': {
+        this.xAxisVelocity = -this.velocity;
+        break;
+      }
+      case 'right': {
+        this.xAxisVelocity = this.velocity;
+        break;
+      }
+      default: {
+        break;
```

Comment:
This is redundant. How is this any more meaningful than not setting the `default`? For ESLint?

Reply:
yeah, I did it because of ESLint. Instead of disabling the rule, I just placed a default? Should I disable ESLint here or keep it in the same way?